vllm/model_executor/layers/sampler.py

```diff
@@ -680,12 +760,36 @@ def _get_logprobs(
     return result_prompt_logprobs, result_sample_logprobs
 
 
+def _modify_greedy_probs_inplace(logprobs: torch.Tensor, probs: torch.Tensor,
```

Comment:
Cade to list how this fits into sampler overall
- Why not use very small temperature instead?